### ISSUE TITLE
[Bug] Batch insertions insert incorrect values

### ISSUE THREAD

**Issue by Pante** [jdbi/jdbi#1987]
In JDBI 3.26.1, batch insertion seem to be broken. It was previously working fine on versions <= 3.26.0.

Given the following Postgres schema.
```sql
CREATE DATABASE sample;

CREATE TABLE foo (
    id uuid PRIMARY KEY,
    next uuid REFERENCES foo(id) DEFERRABLE INITIALLY DEFERRED,

    created_at timestamp NOT NULL,
    updated_at timestamp NOT NULL CHECK (updated_at >= created_at)
);
```

The following snippet of code will insert the wrong values.
```java
var first = UUID.randomUUID();
var second = UUID.randomUUID();

var foos = List.of(
    new Foo(first, second, 0, 2),
    new Foo(second, null, 0, 2)
);
        
jdbi.useTransaction(transaction -> {
    var batch = transaction.prepareBatch("""
        INSERT INTO foo (id, next, created_at, updated_at) VALUES (:id, :next, :created_at, :updated_at)
    """);
    
    for (var foo : foos) {
        batch.bind("id", foo.id())
             .bindBySqlType("next", foo.next(), Types.OTHER)
             .bind("created_at", new Timestamp(foo.createdAt()))
             .bind("updated_at", new Timestamp(foo.updatedAt()))
             .add();
    }
    
    batch.execute();
});
```

![bugg](https://user-images.githubusercontent.com/9427324/148347891-e0985d16-47e6-43b7-9d66-7a304afdf0bb.png)
The value should be `null`, but it's the same as it's ID.

A minimal reproducible example can be found at https://github.com/Pante/jdbi-bug-report.

**Comment by stevenschlansker** [comment]
Thank you for reporting this. We'll evaluate and get back to your shortly, hopefully with a batch release. This is a pretty bad bug to ship...

**Comment by stevenschlansker** [comment]
Thank you @Pante  for reporting this and providing a test case. We've released 3.27.0 which should fix the issue.
